app/Policies/PostPolicy.php

```diff
@@ -50,7 +51,7 @@ public function update(User $user)
      */
     public function delete(User $user)
     {
-        if ($user->can('delete posts')) {
+        if ($user->can('delete any posts') || ($user->can('delete own posts') && $user == auth()->user())){
```

Comment:
The first parameter here is the auth user, so `$user == auth()->user()` will always return true. You need to pass in the post to delete as a second parameter and check if this post belongs to `$user`.